src/rover/main.cpp

```diff
@@ -0,0 +1,174 @@
+//#define DEBUG
+#define TIME
+
+#include "camera.hpp"
+#include "../network/network.hpp"
+
+#include <cstddef>
+#include <iostream>
+#include <vector>
+#include <cstring>
+#include <cstdlib>
+#ifdef TIME
+	#include <chrono> // Time keeping from stdlib
+	using namespace std::chrono;
+#endif
+
+
+const unsigned int CAMERA_MESSAGE_SIZE = 65000;
+
+int main() {
+	/*
+	 * INITILIZATION: Rover Objects
+	 * OPTIMIZE: Consider making a rover class with an iteratable list of cameras
+	 */
+	unsigned int frame_counter = 0;
+	// Open all cameras
+	camera::CaptureSession session0;
+	camera::CaptureSession session1;
+	camera::CaptureSession session2;
+	camera::CaptureSession session3;
+	camera::Error err0 = camera::open(&session0, "/dev/video0", 1280, 720);
+	camera::Error err1 = camera::open(&session1, "/dev/video1", 1280, 720);
+	camera::Error err2 = camera::open(&session2, "/dev/video2", 1280, 720);
+	camera::Error err3 = camera::open(&session3, "/dev/video3", 1280, 720);
+#ifdef DEBUG
+	std::cout << "Camera 0: " << camera::get_error_string(err0) << std::endl;
+	std::cout << "Camera 1: " << camera::get_error_string(err0) << std::endl;
+	std::cout << "Camera 2: " << camera::get_error_string(err0) << std::endl;
+	std::cout << "Camera 3: " << camera::get_error_string(err0) << std::endl;
+#endif
+	// Start cameras
+	err0 = camera::start(&session0);
+	err1 = camera::start(&session1);
+	err2 = camera::start(&session2);
+	err3 = camera::start(&session3);
+#ifdef DEBUG
+	std::cout << "Camera 0: " << camera::get_error_string(err0) << std::endl;
+	std::cout << "Camera 1: " << camera::get_error_string(err1) << std::endl;
+	std::cout << "Camera 2: " << camera::get_error_string(err2) << std::endl;
+	std::cout << "Camera 3: " << camera::get_error_string(err3) << std::endl;
+#endif
+	// Open UDP connection
+	network::Connection conn;
+	network::Error net_err = network::connect(&conn, "192.168.1.1", 45545, 45545);
```

Comment:
Are we sure those numbers/IP make sense? Magic number should be define w/ comments somewhere if possible